tests/fixtures/body-parameter/expected/Resource/TestResource.php

```diff
@@ -17,7 +17,7 @@ class TestResource extends Resource
     public function testSimpleBodyParameter($testString, $parameters = [], $fetch = self::FETCH_OBJECT)
     {
         $queryParam = new QueryParam();
-        $url        = '/test-simple';
+        $url        = 'http://localhost/test-simple';
         $url        = $url . ('?' . $queryParam->buildQueryString($parameters));
         $headers    = array_merge(['Host' => 'localhost'], $queryParam->buildHeaders($parameters));
```

Comment:
> My only concerns in the current behaviour is that by default it only works with the socket client 

It works with all clients look at https://github.com/php-http/client-common/blob/master/src/Plugin/BaseUriPlugin.php